### PR TITLE
[AIRFLOW-5243] Make airflow/configuration pylint compatible

### PR DESCRIPTION
**Pull request by kik-kik**
[AIRFLOW-5243](https://issues.apache.org/jira/browse/AIRFLOW-5243) config and exceptions pylint compliance

### Notes:
```
airflow/configuration.py:199:4: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
airflow/configuration.py:205:4: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
airflow/configuration.py:213:4: R0911: Too many return statements (7/6) (too-many-return-statements)
```

- These 3 errors still remain and I'm not sure how to fix them, too many return statements would require some refactoring but I'm unsure about exactly what is happening in that function.
- I couldn't figure out how to fix the other two

### Description
- As part of [AIRFLOW-4364 Integrate Pylint](https://issues.apache.org/jira/browse/AIRFLOW-4364)

### Tests
- No code changes

### CODE REVIEW

File: airflow/configuration.py [hunk 1]
```diff
@@ -201,7 +208,9 @@ def _get_cmd_option(self, section, key):
         if (section, key) in self.as_command_stdout:
             if super().has_option(section, fallback_key):
                 command = super().get(section, fallback_key)
-                return run_command(command)
```
Comment (ashb) [inline airflow/configuration.py:204]: Why this change? This doesn't look like it would cause a pylint error.

File: airflow/configuration.py [hunk 2]
```diff
@@ -344,6 +353,9 @@ def getsection(self, section):
             _section[key] = val
         return _section
 
+    def _handle_opt(self):
+        pass
```
Comment (ashb) [inline airflow/configuration.py:357]: ⁉️

File: airflow/configuration.py [hunk 3]
```diff
@@ -369,59 +381,86 @@ def as_dict(
...
         ]
 
         for (source_name, config) in configs:
             for section in config.sections():
-                sect = cfg.setdefault(section, OrderedDict())
-                for (k, val) in config.items(section=section, raw=raw):
-                    if display_source:
-                        val = (val, source_name)
-                    sect[k] = val
+                self._set_config_section(config, display_source, raw, section, source_name)
```
Comment (ashb) [inline airflow/configuration.py:392]: Why?

File: airflow/configuration.py (hunk 1 above)
Comment (kik-kik) [reply airflow/configuration.py:204]: Yes, sorry. That was me trying something out, I will revert this change back this weekend.

File: airflow/configuration.py (hunk 3 above)
Comment (kik-kik) [reply airflow/configuration.py:392]: Was trying to reduce the depth (pylint error) by extracting the logic into separate functions

File: airflow/configuration.py (hunk 2 above)
Comment (kik-kik) [reply airflow/configuration.py:357]: Sorry about that, I can't remember what I was thinking at the time. This weekend I will remove it
